rotkehlchen/chain/ethereum/modules/eth2/decoder.py

```diff
@@ -121,62 +127,113 @@ def _decode_eth2_deposit_event(self, context: DecoderContext) -> DecodingOutput:
-        else:
-            source, target = self._get_indexes_or_public_keys(
-                public_keys=[source_public_key, target_public_key],
-            )
-            notes = f'Request to consolidate validator {source} into {target}'
+        if fee_event is None:
+            log.warning(f'Failed to find fee event in ETH2 request transaction {context.transaction}')  # noqa: E501
 
-        return DecodingOutput(event=self.base.make_event_next_index(
+        return fee_event, self.base.make_event_next_index(
             tx_hash=context.transaction.tx_hash,
             timestamp=context.transaction.timestamp,
             event_type=HistoryEventType.INFORMATIONAL,
             event_subtype=HistoryEventSubType.NONE,
             asset=A_ETH,
             amount=ZERO,
-            location_label=caller_address,
-            notes=notes,
+            location_label=bytes_to_address(b'\x00' * 12 + context.tx_log.data[:20]),
             address=context.tx_log.address,
             counterparty=CPT_ETH2,
-        ))
+        )
+
+    def _decode_eth2_consolidation_request(self, context: DecoderContext) -> DecodingOutput:
+        """Decode a request to consolidate validators.
+        Handles two types of requests:
+        - Self consolidation - changes the withdrawal credentials to 0x02 (accumulating validator)
+          See https://epf.wiki/?ref=blog.obol.org#/wiki/pectra-faq?id=q-i-have-a-validator-with-0x01-credentials-how-do-i-move-to-0x02
+        - Normal consolidation - consolidates the source validator into the target validator
+        """
+        if (events := self._decode_eth2_request_with_fee(
+            context=context,
+            expected_data_length=116,
+        )) is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        fee_event, info_event = events
+        source_public_key = Eth2PubKey(bytes_to_hexstr(context.tx_log.data[20:68]))
+        target_public_key = Eth2PubKey(bytes_to_hexstr(context.tx_log.data[68:116]))
+        if source_public_key == target_public_key:
+            target = self._get_indexes_or_public_keys(public_keys=[target_public_key])[0]
+            info_event.notes = f'Request to convert validator {target} into an accumulating validator'  # noqa: E501
+        else:
+            source, target = self._get_indexes_or_public_keys(
+                public_keys=[source_public_key, target_public_key],
+            )
+            info_event.notes = f'Request to consolidate validator {source} into {target}'
+
+        if fee_event is not None:
+            fee_event.notes = f'Spend {fee_event.amount} ETH as validator consolidation fee'
+
+        return DecodingOutput(event=info_event)
+
+    def _decode_eth2_withdrawal_request(self, context: DecoderContext) -> DecodingOutput:
+        """Decode a withdrawal/exit request.
+        If amount is specified, it is a partial withdrawal request.
+        If amount is zero, it indicates a request to fully exit the validator.
```

Comment:
you might need to reword this differently because `amount` is always present from what i can read. something like "if amount > 0", "if amount equals 0"